### PR TITLE
docs: fix errors in the bot example and add intent

### PR DESCRIPTION
**Pull request by rdni**
## Pull Request Type

- [ ] Feature addition
- [ ] Bugfix
- [X] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Changed the example bot to make it function correctly as well as have up to date information on the docs

## Changes

- Changed the docs link to be up to date
- Added the message content intent.
- Changed the parameter for the message listen from "interactions.Message" to "interactions.api.events.MessageCreate".

## Test Scenarios

No tests are required as it is not changing the library it self

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [X] I've ensured my code works on Python `3.11.x`

## Checklist

- [ ] I've run the `pre-commit` code linter over all edited files
- [X] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

### CODE REVIEW

File: examples/bot.py [hunk 1]
```diff
@@ -25,18 +28,20 @@ async def on_ready():
     print(f"Our latency is {round(client.latency)} ms.")
 
 
+# We can either pass in the event name or make the function name be the event name.
 @interactions.listen("on_message_create")
-async def name_this_however_you_want(message: interactions.Message):
+async def name_this_however_you_want(message_create: interactions.api.events.MessageCreate):
```
Comment (AstreaTSS) [inline examples/bot.py]: This can just be `interactions.events.MessageCreate`.

File: examples/bot.py [hunk 2]
```diff
@@ -10,7 +10,10 @@
 # When you make a bot, we refer to it as the "client."
 # The client is the main object that interacts with the Gateway, what talks to Discord.
 # The client is also the main object that interacts with the API, what makes requests with Discord.
-client = interactions.Client()
+# The client can also have "intents" that are what the bot recieves,
+# in this case the default ones and message content
+intents = interactions.Intents.DEFAULT | interactions.Intents.MESSAGE_CONTENT
```
Comment (AstreaTSS) [inline examples/bot.py:16]: Please add a note that message content is a special intent that requires toggling it on in the dashboard too.

**Review by AstreaTSS — COMMENTED** [review]
Overall a good start - just need a couple of wording adjustments (and one code change).

File: examples/bot.py [hunk 3]
```diff
@@ -66,5 +71,6 @@ async def hello_world(ctx: interactions.SlashContext):
 #   object on line 13.
 # - we are not setting a presence.
 # - we are not automatically sharding, and registering the connection under 1 shard.
-# - we are using default intents, which are Gateway intents excluding privileged ones.
+# - we are using default intents and the message content intent, which are Gateway intents and
+# - message content excluding privileged ones.
```
Comment (AstreaTSS) [inline examples/bot.py]: This is... confusing. It makes it sound like message content *isn't* a privileged intent - this sentence may need a total rewording to make more sense.

Comment (rdni) [reply examples/bot.py]: I reworded the comments to try and make it make more sense, and explicitly say that message content is a privileged intent

File: examples/bot.py [hunk 4]
```diff
@@ -25,18 +29,20 @@ async def on_ready():
...
 @interactions.listen("on_message_create")
-async def name_this_however_you_want(message: interactions.Message):
+async def name_this_however_you_want(message_create: interactions.events.MessageCreate):
     # Whenever we specify any other event type that isn't "READY," the function underneath
     # the decorator will most likely have an argument required. This argument is the data
     # that is being supplied back to us developers, which we call a data model.
 
     # In this example, we're listening to messages being created. This means we can expect
-    # a "message" argument to be passed to the function, which will be the data model of such.
+    # a "message_create" argument to be passed to the function, which will be the data model of such.
```
Comment (silasary) [inline examples/bot.py]: Message_create is an event model, not the actual message data model

Comment (rdni) [reply examples/bot.py]: I have changed the comment for this.

**Review by silasary — APPROVED** [review]

**Pull request merged**